### PR TITLE
Only use pinned thread mode in Spark 3.2 when explicitly enabled

### PR DESCRIPTION
**Pull request by WeichenXu123**
Only use pinned thread mode in Spark 3.2 when explicitly enabled
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

### CODE REVIEW

File: joblibspark/backend.py [hunk 1]
```diff
@@ -86,11 +97,10 @@ def effective_n_jobs(self, n_jobs):
             # n_jobs=-1 means requesting all available workers
             n_jobs = max_num_concurrent_tasks
         if n_jobs > max_num_concurrent_tasks:
-            warnings.warn("User-specified n_jobs ({n}) is greater than the max number of "
-                          "concurrent tasks ({c}) this cluster can run now. If dynamic "
-                          "allocation is enabled for the cluster, you might see more "
-                          "executors allocated."
-                          .format(n=n_jobs, c=max_num_concurrent_tasks))
+            warnings.warn(f"User-specified n_jobs ({n_jobs}) is greater than the max number of "
```
Comment (dbczumar) [inline joblibspark/backend.py:100]: Nice f-string :D

File: joblibspark/backend.py [hunk 2]
```diff
@@ -131,35 +141,40 @@ def apply_async(self, func, callback=None):
         # Note the `func` args is a batch here. (BatchedCalls type)
         # See joblib.parallel.Parallel._dispatch
 
-        spark_major_minor_ver = \
-            VersionUtils.majorMinorVersion(pyspark.__version__)
-        if spark_major_minor_ver < (3, 2):
-            def run_on_worker_and_fetch_result():
-                # TODO: handle possible spark exception here. # pylint: disable=fixme
```
Comment (dbczumar) [inline joblibspark/backend.py:138]: Do we need to do any additional exception handling before removing this TODO?

**Review by dbczumar — APPROVED** [review]
LGTM with one question about exception handlings

File: joblibspark/backend.py (hunk 2 above)
Comment (WeichenXu123) [reply joblibspark/backend.py:138]: Good catch. I add it back.

**Pull request merged**